src/java/com/rapleaf/jack/transaction/TransactorImpl.java

```diff
@@ -0,0 +1,198 @@
+package com.rapleaf.jack.transaction;
+
+import java.util.concurrent.Callable;
+
+import com.google.common.base.Preconditions;
+import org.joda.time.Duration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rapleaf.jack.IDb;
+import com.rapleaf.jack.exception.SqlExecutionFailureException;
+
+/**
+ * If there is any exception while executing the query, throws {@link com.rapleaf.jack.exception.SqlExecutionFailureException}.
+ * <p>
+ * If there is no available connections, throws {@link com.rapleaf.jack.exception.NoAvailableConnectionException}.
+ * Users can either increase the max total connections or max wait time.
+ * <p>
+ * If the DB manager has already been closed, throws {@link IllegalStateException}.
+ * <p>
+ * If new DB connections cannot be created, throws {@link com.rapleaf.jack.exception.ConnectionCreationFailureException}.
+ */
+public class TransactorImpl<DB extends IDb> implements ITransactor<DB> {
+  private static final Logger LOG = LoggerFactory.getLogger(TransactorImpl.class);
+
+  private final IDbManager<DB> dbManager;
+
+  TransactorImpl(IDbManager<DB> dbManager) {
+    this.dbManager = dbManager;
+  }
+
+  public static <DB extends IDb> Builder<DB> create(Callable<DB> dbConstructor) {
+    return new Builder<>(dbConstructor);
+  }
+
+  @Override
+  public <T> T execute(IQuery<DB, T> query) {
+    return execute(query, false);
+  }
+
+  @Override
+  public <T> T executeAsTransaction(IQuery<DB, T> query) {
+    return execute(query, true);
+  }
+
+  @Override
+  public void execute(IExecution<DB> execution) {
+    execute(execution, false);
+  }
+
+  @Override
+  public void executeAsTransaction(IExecution<DB> execution) {
+    execute(execution, true);
+  }
+
+  private <T> T execute(IQuery<DB, T> query, boolean asTransaction) {
+    DB connection = dbManager.getConnection();
+    connection.setAutoCommit(!asTransaction);
```

Comment:
@tuliren where does autocommit get set back to it's default value (true)? When the connection is returned to the pool it could presumably still have autocommit disabled, right?